base/div.jl

```diff
@@ -133,15 +133,30 @@ julia> divrem(7,3)
 ```
 """
 divrem(x, y) = divrem(x, y, RoundToZero)
+
+#avoids calling rem for Integers-Integers,
+#a-d*b not precise for Floats - AbstractFloat, AbstractIrrational. Rationals are still slower
+function divrem(a::Integer, b::Integer, r::RoundingMode)
```

Comment:
I think this should use explicit types on `r` for the two special cases, together with the note to remove them in 2.0.